Provis.Core/Services/WorkspaceService.cs

```diff
@@ -343,18 +359,23 @@ public async Task<List<WorkspaceMemberDTO>> GetWorkspaceMembersAsync(int workspa
 
         public async Task DeleteFromWorkspaceAsync(int workspaceId, string userId)
         {
-            var user = await _userManager.FindByIdAsync(userId);
+            var userWorkspSpecification = new UserWorkspaces.WorkspaceMember(userId, workspaceId);
+            var userWorksp = await _userWorkspaceRepository.GetFirstBySpecAsync(userWorkspSpecification);
 
-            var specification = new UserWorkspaces.WorkspaceMember(userId, workspaceId);
-            var userWorksp = await _userWorkspaceRepository.GetFirstBySpecAsync(specification);
+            if (userWorksp.RoleId == 1)
+            {
+                throw new HttpException(System.Net.HttpStatusCode.NotFound,
+                    "Owner can't leave workspace");
+            }
 
-            var userTasks = user.UserTasks.Where(o => o.Task.WorkspaceId == workspaceId);
+            var userTaskSpecification = new UserTasks.UserTaskList(userId, workspaceId);
+            var userTasks = await _userTaskRepository.GetListBySpecAsync(userTaskSpecification);
 
             if (userTasks != null)
             {
                 foreach (var userTask in userTasks)
                 {
-                    userTask.IsUserDeleted = true;
+                    userTask.Item2.IsUserDeleted = true;
                 }
```

Comment:
bool already has a false value, should we set it again ?

Reply:
when we deleting user from workspace, we need to mark all user task as "is deleted = true"